nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,480 @@
+        a list of seperated regions with each region lying on a 3D volume
+        concatenated into a 4D Nifti like object.
+
+    `index_`: a numpy array
+        an array of list of indices where each index value is assigned to
+        each seperate region of its corresponding family of brain maps.
+
+    `signals_`: a numpy array
+        a list of averaged timeseries signals of the subjects extracted from
+        each region.
+
+    References
+    ----------
+    * Abraham et al. "Region segmentation for sparse decompositions: better
+      brain parcellations from rest fMRI", Sparsity Techniques in Medical Imaging,
+      Sep 2014, Boston, United States. pp.8
+    """
+    def __init__(self, maps_img, mask=None, target_affine=None,
+                 target_shape=None, standardize=False, low_pass=None,
+                 high_pass=None, t_r=None, memory=Memory(cachedir=None),
+                 min_size=20, thresholding_strategy='ratio_n_voxels',
+                 threshold='auto', extractor='connected_components',
+                 peak_local_smooth=6., verbose=0):
+        self.maps_img = maps_img
+        self.mask = mask
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.standardize = standardize
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.memory = memory
+
+        # parameters for region extraction
+        self.min_size = min_size
+        self.thresholding_strategy = thresholding_strategy
+        self.threshold = threshold
+        self.extractor = extractor
+        self.peak_local_smooth = peak_local_smooth
+
+        self.verbose = verbose
+
+    def fit(self, X=None, y=None):
+        """ Prepare or set up the data for the region extraction
+
+        """
+        self.maps_img_ = check_niimg(self.maps_img)
+
+        if isinstance(self.mask, NiftiMasker):
+            self.masker_ = clone(self.mask)
+        else:
+            self.masker_ = NiftiMasker(mask_img=self.mask,
+                                       target_affine=self.target_affine,
+                                       target_shape=self.target_shape,
+                                       standardize=False,
+                                       low_pass=self.low_pass,
+                                       high_pass=self.high_pass,
+                                       mask_strategy='background',
+                                       t_r=self.t_r,
+                                       memory=self.memory)
+
+        if self.masker_.mask_img is None:
+            self.masker_.fit(self.maps_img_)
+        else:
+            self.masker_.fit()
+        self.mask_img_ = self.masker_.mask_img_
+
+        parameters = get_params(NiftiMasker, self)
+        parameters['detrend'] = True
+        # foreground extraction of input data "maps_img"
+        self.threshold_maps_img_ = estimate_apply_threshold_to_maps(
```

Comment:
I don't understand the pattern here: what is the purpose of fit? What is the purpose of transform? Shouldn't that be linked to NiftiMapsMasker somehow?